src/lib/qnio/nio_client.c

```diff
@@ -590,6 +614,37 @@ qnc_channel_open(void *channel_arg)
 
     netch = (struct network_channel *)malloc(sizeof (struct network_channel));
     memset(netch, 0, sizeof (struct network_channel));
+    channel = &netch->channel;
+
+    /*
+     * Initialize SSL context for the new channel based on the certs and keys
+     * passed by the user.
+     */
+    if (!is_secure()) {
```

Comment:
is_secure() tests for existence of a magic directory on disk.  Now we're passing the certificate paths into the iio_open() method explicitly, continuing to use this is_secure() method is a bad idea - it causes the requested config to be silently ignored. Instead of checking is_secure(), this could just do  "cacert == NULL" - ie providing an non-NULL cacert to iio_open() should turn on use of TLS